src/uct/ib/ud/base/ud_ep.c

```diff
@@ -665,23 +677,37 @@ static uct_ud_ep_t *uct_ud_ep_create_passive(uct_ud_iface_t *iface, uct_ud_ctl_h
     /* create new endpoint */
     params.field_mask = UCT_EP_PARAM_FIELD_IFACE;
     params.iface      = &iface->super.super.super;
-    status = uct_ep_create(&params, &ep_h);
-    ucs_assert_always(status == UCS_OK);
+    status            = uct_ep_create(&params, &ep_h);
+    if (status != UCS_OK) {
+        return NULL;
+    }
+
     ep = ucs_derived_of(ep_h, uct_ud_ep_t);
 
     status = uct_ep_connect_to_ep(ep_h, (void*)uct_ud_creq_ib_addr(ctl),
                                   (void*)&ctl->conn_req.ep_addr);
-    ucs_assert_always(status == UCS_OK);
+    if (status != UCS_OK) {
+        goto err_disconnect;
```

Comment:
going to disconnect looks strange when connect fails

Reply:
fixed naming